File: CB.EN.U4CYS22057/30_03_2023/python/bubble_sort.py
# Declaring variables
# Getting input for size of array and creating array
print("Enter the number of elements in the array")
size = int(input())
array = [0] * (size)

# Getting input for values of array
for i in range(0, size - 1 + 1, 1):
    print("Enter the value for index " + str(i))
    array[i] = int(input())
for i in range(0, size - 1 + 1, 1):
    for j in range(0, size - 2 - i + 1, 1):
        
        # Checking if a value is greater than its next value
        if array[j] > array[j + 1]:
            
            # exchanging values
            a = array[j]
            array[j] = array[j + 1]
            array[j + 1] = a
    
    # Printing output after each pass
    print("The array after pass " + str(i + 1) + "  is")
    for j in range(0, size - 1 + 1, 1):
        print(array[j])

File: CB.EN.U4CYS22057/30_03_2023/c/bubble_sort.c
#include <stdio.h>

int main() {
    // Getting input for size of array and creating array
    printf("Enter the number of elements in the array:\n");
    int size;
    scanf("%d", &size);
    int array[size];

    // Getting input for values of array
    for (int i = 0; i < size; i++) {
        printf("Enter the value for index %d:\n", i);
        scanf("%d", &array[i]);
    }

    // Sorting array using bubble sort
    for (int i = 0; i < size-1; i++) {
        for (int j = 0; j < size-1-i; j++) {
            // Checking if a value is greater than its next value
            if (array[j] > array[j+1]) {
                // exchanging values
                int a = array[j];
                array[j] = array[j+1];
                array[j+1] = a;
            }
        }

        // Printing output after each pass
        printf("The array after pass %d is:\n", i+1);
        for (int j = 0; j < size; j++) {
            printf("%d\n", array[j]);
        }
    }

    return 0;
}
